targeteditor/src/main/scala/explore/targeteditor/SkyPlot.scala

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.targeteditor
+
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+import scala.collection.immutable.HashSet
+import scala.scalajs.js
+
+import cats.implicits._
+import explore.implicits._
+import explore.model.reusability._
+import gem.enum.Site
+import gem.math.ObservingNight
+import gem.util.Enumerated
+import gpp.highcharts.highchartsStrings.line
+import gpp.highcharts.mod.XAxisLabelsOptions
+import gpp.highcharts.mod._
+import gpp.ui.reusability._
+import gsp.math.Angle
+import gsp.math.Coordinates
+import gsp.math.skycalc.TwilightBoundType
+import japgolly.scalajs.react.MonocleReact._
+import japgolly.scalajs.react._
+import monocle.macros.Lenses
+import react.common._
+import react.highcharts.Chart
+import shapeless._
+
+import js.JSConverters._
+
+final case class SkyPlot(
+  site:   Site,
+  coords: Coordinates,
+  date:   LocalDate,
+  zoneId: ZoneId,
+  height: Int
+) extends ReactProps[SkyPlot](SkyPlot.component)
+
+object SkyPlot {
+  type Props = SkyPlot
+
+  @Lenses
+  case class State(shownSeries: HashSet[ElevationSeries] = HashSet(ElevationSeries.Elevation))
+
+  implicit private val propsReuse: Reusability[Props] = Reusability.derive
+  // State doesn't trigger rerenders. We keep track of what is shown in case there is a
+  // rerender due to a change of properties.
+  implicit private val stateReuse: Reusability[State] = Reusability.always
+
+  private val PlotEvery: Duration   = Duration.ofMinutes(1)
+  private val MillisPerHour: Double = 60 * 60 * 1000
+
+  protected case class SeriesData(
+    targetAltitude:  List[Chart.Data],
+    skyBrightness:   List[Chart.Data],
+    parallaticAngle: List[Chart.Data],
+    moonAltitude:    List[Chart.Data]
+  )
+
+  sealed abstract class ElevationSeries(
+    val name:  String,
+    val yAxis: Int,
+    val data:  SeriesData => List[Chart.Data]
+  )
+  object ElevationSeries extends Enumerated[ElevationSeries] {
+    case object Elevation        extends ElevationSeries("Elevation", 0, _.targetAltitude)
+    case object ParallacticAngle extends ElevationSeries("Parallatic Angle", 1, _.parallaticAngle)
+    case object SkyBrightness    extends ElevationSeries("Sky Brightness", 2, _.skyBrightness)
+    case object LunarElevation   extends ElevationSeries("Lunar Elevation", 0, _.moonAltitude)
+
+    def tag(a: ElevationSeries) = a.name
+
+    val all = List(Elevation, ParallacticAngle, SkyBrightness, LunarElevation)
+  }
+
+  private val seriesDataGen = Generic[SeriesData]
+
+  val dateTimeFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm")
+
+  def formatAngle(degs: Double): String = {
```

Comment:
Can/should we use a `Formatter` from the angle class?

Reply:
The existing fomatter: a) Uses `:` to separate all components instead of the symbols for degress, arcminutes and arcseconds. b) Doesn't round the decimals, showing quite a bit of them. It's appropriate for exact, human readable, string representation but for UI it wasn't looking so nice.

Reply:
Ok, we may want to move these (later) to either math or `gpp-ui` to share them across projects

Reply:
That's a good idea.